### PR TITLE
Feature/aks

### PR DESCRIPTION
**Pull request by MA3CIN**
I still need to:
Adjust the ENVS for docker compose (make user and psswd as envs passed to containers in compose)
Merge Mateusz's MySQL script part with direct VNET to AKS
install the mysql tool for script purposes

dynamically populate the SecretProvider with tenant id, keyvault name (can stay) + AKS tenant id...?

### CODE REVIEW

File: aks/push_to_acr.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+docker build -t resilientsixregistry.azurecr.io/geo-service:v1.0.3 ../services/geo-service/.
+docker push resilientsixregistry.azurecr.io/geo-service:v1.0.3
+
+# infra service
+docker build -t resilientsixregistry.azurecr.io/infra-service:v1.0.3 ../services/infra-service/.
+docker push resilientsixregistry.azurecr.io/infra-service:v1.0.3
+
+# stats service
+docker build -t resilientsixregistry.azurecr.io/stats-service:v1.0.3 ../services/stats-service/.
+docker push resilientsixregistry.azurecr.io/stats-service:v1.0.3
```
Comment (jgorczynski) [inline aks/push_to_acr.sh]: Minor formatting remark - no new line at EOF

File: aks/yaml/stats-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    matchLabels:
+      app: stats-deploy
+  template:
+    metadata:
+      labels:
+        app: stats-deploy
+    spec:
+      containers:
+        - name: stats
+          # docker.io/marcinziolkowski/stats-service:v1.0.1
```
Comment (jgorczynski) [inline aks/yaml/stats-deploy.yaml]: Is this comment necessary?

**Review by jgorczynski — CHANGES_REQUESTED** [review]
Minor formatting remark, nothing else so far

File: aks/yaml/stats-deploy.yaml (hunk 2 above)
Comment (MA3CIN) [reply aks/yaml/stats-deploy.yaml]: i fixed this on some other branch - some time ago docker defaulted to a different image registry, this is no longer needed

File: aks/mysql_create.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#   --resource-group resilientSixGroup \
+#   --name resilientsix-mysql-db \
+#   --database-name cache \
+#   --admin-user admin_user \
+#   --admin-password admin_password \
+# 	--location polandcentral \
+# 	--tier GeneralPurpose \
+# 	--sku-name Standard_D2ads_v5 \
+# 	--version 8.0.21 \
+#   --yes
```
Comment (WasilewM) [inline aks/mysql_create.sh]: we can delete unnecessary code

File: aks/windows/mysql_create.cmd [hunk 4]
```diff
@@ -0,0 +1 @@
+az mysql flexible-server create  --resource-group resilientSixGroup  --name resilientsix-mysql-db  --database-name cache  --admin-user admin_user --admin-password admin_password --location polandcentral 	--tier GeneralPurpose 	--sku-name Standard_D2ds_v4 	--version 8.0.21   --yes
```
Comment (WasilewM) [inline aks/windows/mysql_create.cmd:1]: we need to create `devices`, `geolocation` and `services databases. Not sure whether we need the `cache` DB here

File: kubernetes/geo-deploy.yaml [hunk 5]
```diff
@@ -25,10 +25,10 @@ spec:
           resources:
             limits:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
             requests:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
```
Comment (WasilewM) [inline kubernetes/geo-deploy.yaml:33]: Why do we lower the reqested cpu value?

File: kubernetes/infra-deploy.yaml [hunk 6]
```diff
@@ -25,10 +25,10 @@ spec:
           resources:
             limits:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
             requests:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
```
Comment (WasilewM) [inline kubernetes/infra-deploy.yaml:31]: Why do we lower the requested cpu value?

File: kubernetes/stats-deploy.yaml [hunk 7]
```diff
@@ -26,10 +26,10 @@ spec:
           resources:
             limits:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
             requests:
               memory: "128Mi"
-              cpu: "500m"
+              cpu: "250m"
 ---
```
Comment (WasilewM) [inline kubernetes/stats-deploy.yaml:34]: Why do we lower the requested cpu value?

**Review by WasilewM — COMMENTED** [review]
+1

File: aks/windows/mysql_create.cmd (hunk 4 above)
Comment (MA3CIN) [reply aks/windows/mysql_create.cmd:1]: i'm not sure what you mean. Cache db?

File: kubernetes/geo-deploy.yaml (hunk 5 above)
Comment (MA3CIN) [reply kubernetes/geo-deploy.yaml:33]: Azure student subscription is not enough for 3x500m. 250m is enough for our needs, even 100 would be enough

File: kubernetes/infra-deploy.yaml (hunk 6 above)
Comment (MA3CIN) [reply kubernetes/infra-deploy.yaml:31]: Azure student subscription is not enough for 3x500m. 250m is enough for our needs, even 100 would be enough

File: kubernetes/stats-deploy.yaml (hunk 7 above)
Comment (MA3CIN) [reply kubernetes/stats-deploy.yaml:34]: Azure student subscription is not enough for 3x500m. 250m is enough for our needs, even 100 would be enough

File: aks/mysql_create.sh [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+az mysql flexible-server db create   --resource-group resilientSixGroup   --server-name resilientsix-mysql-db2   --database-name devices
+
+az mysql flexible-server db create   --resource-group resilientSixGroup   --server-name resilientsix-mysql-db2  --database-name statistics
+
+
+# mysql -u admin_user -padmin_password -h resilientsix-mysql-db2.mysql.database devices
+
+# source databases/devices_db_schema.sql
+# source databases/devices_sample_data.sql
```
Comment (jgorczynski) [inline aks/mysql_create.sh:17]: minor formatting remark - no new line at EOF

File: aks/akv_create.sh [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+# RBAC is a pain in the ass unfortunately :)
+
+
+az keyvault create --name resilientsixkeyvault2 --resource-group resilientSixGroup --location polandcentral
+
+az keyvault secret set --vault-name resilientsixkeyvault2 --name DB-URL --value resilientsix-mysql-db2.mysql.database.azure.com
+
+az keyvault secret set --vault-name resilientsixkeyvault2 --name DB-USER --value admin_user
+
+az keyvault secret set --vault-name resilientsixkeyvault2 --name DB-PSWD --value admin_password
```
Comment (jgorczynski) [inline aks/akv_create.sh:11]: minor formatting remark - no new line at EOF

File: aks/push_to_acr.sh [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+# PARAMETER = TAG of image
+
+docker build -t resilientsixregistry.azurecr.io/geo-service:v2.0.3 ../../services/geo-service/.
+docker push resilientsixregistry.azurecr.io/geo-service:v2.0.3
+
+docker build -t resilientsixregistry.azurecr.io/infra-service:v2.0.3 ../../services/infra-service/.
+docker push resilientsixregistry.azurecr.io/infra-service:v2.0.3
+
+docker build -t resilientsixregistry.azurecr.io/stats-service:v2.0.3 ../../services/stats-service/.
+docker push resilientsixregistry.azurecr.io/stats-service:v2.0.3
```
Comment (jgorczynski) [inline aks/push_to_acr.sh:12]: minor formatting remark - no new line at EOF

File: aks/windows/aks_create.cmd [hunk 11]
```diff
@@ -0,0 +1,5 @@
+az aks create -n resilientSixCluster -g resilientSixGroup --generate-ssh-keys --attach-acr resilientsixregistry --node-count 2 --enable-managed-identity --enable-addons azure-keyvault-secrets-provider --enable-addons open-service-mesh
+
+az aks enable-addons --addons azure-keyvault-secrets-provider --name resilientSixCluster --resource-group resilientSixGroup
+
+az aks get-credentials --resource-group resilientSixGroup --name resilientSixCluster
```
Comment (jgorczynski) [inline aks/windows/aks_create.cmd:5]: minor formatting remark - no new line at EOF

File: aks/windows/akv_create.cmd [hunk 12]
```diff
@@ -0,0 +1,7 @@
+az keyvault create --name resilientsixkeyvault3 --resource-group resilientSixGroup --location polandcentral
+
+az keyvault secret set --vault-name resilientsixkeyvault3 --name DB-URL --value resilientsix-mysql-db.mysql.database
+
+az keyvault secret set --vault-name resilientsixkeyvault3 --name DB-USER --value admin_user
+
+az keyvault secret set --vault-name resilientsixkeyvault3 --name DB-PSWD --value admin_password
```
Comment (jgorczynski) [inline aks/windows/akv_create.cmd:7]: minor formatting remark - no new line at EOF

File: aks/windows/push_to_acr.cmd [hunk 13]
```diff
@@ -0,0 +1,8 @@
+docker build -t resilientsixregistry.azurecr.io/geo-service:v2.0.3 ../../services/geo-service/.
+docker push resilientsixregistry.azurecr.io/geo-service:v2.0.3
+
+docker build -t resilientsixregistry.azurecr.io/infra-service:v2.0.3 ../../services/infra-service/.
+docker push resilientsixregistry.azurecr.io/infra-service:v2.0.3
+
+docker build -t resilientsixregistry.azurecr.io/stats-service:v2.0.3 ../../services/stats-service/.
+docker push resilientsixregistry.azurecr.io/stats-service:v2.0.3
```
Comment (jgorczynski) [inline aks/windows/push_to_acr.cmd:8]: minor formatting remark - no new line at EOF

**Review by jgorczynski — COMMENTED** [review]
Mainly formatting remarks

**Review by jgorczynski — APPROVED** [review]

**Review by WasilewM — APPROVED** [review]

**Pull request merged**